### PR TITLE
Results Panes Styling

### PR DESCRIPTION
**Pull request by christinem**
With @reid and @david-yz-liu 's suggestions, I've made changes to the styling of the results panes for markers and students. 

The results view code is not as clean as it could be, as I was working with what had been done before. It could use a lot of refactoring.

### CODE REVIEW

**Comment by reid** [comment]
Hi! I didn't make any suggestions, perhaps you meant another Reid?

**Comment by christinem** [comment]
@reid my mistake! I meant @reidka.

File: app/assets/javascripts/Grader/marking.js [hunk 1]
```diff
@@ -167,9 +169,13 @@ function update_rubric_mark(elem, mark_id, value) {
...
       select_mark(mark_id, value);
-      document.getElementById('mark_criterion_title_' + mark_id + '_mark')
-              .innerHTML = elem.innerHTML;
       update_total_mark(total);
+        
+      var div = document.createElement('div');
+      div.innerHTML = elem.innerHTML
+      var titleLevel = jQuery(div).find('strong').text();
+      document.getElementById('mark_criterion_title_' + mark_id + '_level')
+          .innerHTML = titleLevel;
```
Comment (david-yz-liu) [inline app/assets/javascripts/Grader/marking.js]: Align the periods

File: app/views/results/common/_flexible_criterion.html.erb [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
-     data-id='<%= mark_criterion.id %>'>
-  <% if !mark.mark.nil? -%>
-    <strong><%= mark.mark %>&nbsp;/ <%= mark_criterion.max %></strong>
-    <p><%= mark_criterion.description %></p>
-  <% end -%>
+<div id="<%="mark_criterion_title_#{mark_criterion.id}_old_mark"%>"
+     class="mark_criterion_old_mark float_left">
+  <strong><%= if(!old_mark.nil? && !old_mark.mark.nil?) then
+                "#{I18n.t('marker.marks.old')}#{old_mark.mark.to_s}"
+              else "" end %></strong>
```
Comment (david-yz-liu) [inline app/views/results/common/_flexible_criterion.html.erb]: Use single-quotes when possible, and put the closing tag on a new line

File: app/views/results/common/_flexible_criterion.html.erb [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
-<div id="<%= 'mark_criterion_title_#{mark.id}_mark' %>"
-     class='padding_left mark_description'
-     data-id='<%= mark_criterion.id %>'>
-  <% if !mark.mark.nil? -%>
-    <strong><%= mark.mark %>&nbsp;/ <%= mark_criterion.max %></strong>
-    <p><%= mark_criterion.description %></p>
-  <% end -%>
+<div id="<%="mark_criterion_title_#{mark_criterion.id}_old_mark"%>"
+     class="mark_criterion_old_mark float_left">
+  <strong><%= if(!old_mark.nil? && !old_mark.mark.nil?) then
```
Comment (david-yz-liu) [inline app/views/results/common/_flexible_criterion.html.erb]: You don't need parentheses around the condition

File: app/views/results/marker/_marker_flexible_criterion_li.html.erb [hunk 4]
```diff
@@ -7,33 +7,39 @@
...
+           class='flexible_criterion_title_div_level float_left'
+           onclick="<%="focus_mark_criterion(#{mark_criterion.id})"%>">
+        <strong><%= (mark_criterion.name) %></strong>
+      </div>
+
+      <div id="mark_flexible_<%=mark.id%>" class="mark_flexible">
+        <%= text_field_tag('mark_' + mark.id.to_s, mark.mark,
+                           size: 4,
+                           class: 'mark_grade_input',
+                           data: {:id => mark_criterion.id},
```
Comment (david-yz-liu) [inline app/views/results/marker/_marker_flexible_criterion_li.html.erb]: Use the new hash syntax `id: ...`

File: app/views/results/marker/_marker_flexible_criterion_li.html.erb [hunk 5]
```diff
@@ -7,33 +7,39 @@
...
+        /&nbsp; <span id="mark_flexible_max_<%=mark.id%>"><%= mark_criterion.max.to_s%> </span>
+      </div>
+
+      <div class="mark_description" style="display: none;" data-id='<%= mark_criterion.id %>'>
+        <%= mark.mark %>
+      </div>
+
+      <div id="<%="criterion_info_#{mark_criterion.id}"%>" class="flexible_criterion_info">
+        <div id="<%="mark_criterion_inputs_#{mark_criterion.id}"%>">
+          <%=mark_criterion.description %>
```
Comment (david-yz-liu) [inline app/views/results/marker/_marker_flexible_criterion_li.html.erb]: space after `=`

**Comment by david-yz-liu** [comment]
Make the actual mark numbers float right (both rubric and flexible). Put the level name in parentheses (rubric).

I don't think the class `float_left` exists any more, but you should check that.

Decide what needs ids and what doesn't (most things really don't), and remove the unnecessary ids.

**Comment by david-yz-liu** [comment]
Oh, and please make the `+` and `-` aligned properly with the title for each criterion (right now it's still a bit too high).

**Comment by christinem** [comment]
@david-yz-liu @reid This should finally be ready for another full review.

**Comment by reidka** [comment]
@christinem: did you push your latest changes?  I'm only seeing commits from 14 days ago.

**Comment by christinem** [comment]
@reidka odd, yeah I've pushed my latest ones, I can see them all on my side in the commits tab of this PR.

**Comment by david-yz-liu** [comment]
Thanks, @christinem. I have merged this in. There are still two areas of improvement:

1) For remark requests with flexible criteria, the "Old Mark" will always appear when the input text changes, even if the same number is retyped. (E.g., if a mark is shown as 1.0, and I delete the 0 and retype it, MarkUs will consider this a change.) The value of the box should be checked and compared with the old value - this could be a server-side action if that's easier.
2) Too many classes and ids still are specific to the two types of criteria. There should be exactly one such class, and all the other CSS and Javascript should use selectors based on that one class.

**Pull request merged**